authorization-server/src/modules/auth/roles-validation.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { LoggerService } from '../logger';
+import { IDidAccessTokenPayload } from './types';
+
+@Injectable()
+export class RolesValidationService {
+  constructor(
+    private readonly configService: ConfigService,
+    private readonly logger: LoggerService,
+  ) {
+    this.logger.setContext(RolesValidationService.name);
+  }
+
+  public async didAccessTokenRolesAreValid(
+    verifiedRoles: IDidAccessTokenPayload['verifiedRoles'],
+  ): Promise<boolean> {
+    const acceptedRoles =
+      this.configService.get('ACCEPTED_ROLES')?.split(',') || [];
+    const roles = verifiedRoles.map((r) => r.namespace);
+
+    this.logger.debug(
+      `validating ${JSON.stringify(roles)} against ${acceptedRoles}`,
+    );
+
+    return roles.some((r) => acceptedRoles.includes(r));
```

Comment:
I think it would be worth testing testing the behaviour if the `ACCEPTED_ROLES` string is empty. 

For instance, this code `[1].some(v => [].includes(v))` would return `false`, so I think if `ACCEPTED_ROLES` is empty, then the return value might always be false, which I'm not sure we want

Reply:
The application will not start with accepted roles empty, but this service does not know about it :). I am also not 100% sure, but it is not possible to log in when there are no accepted roles set AFAIR.

Reply:
Just checked - it is possible to log in when the ACCEPTED_ROLES string is empty, but it is required to set anything, to make the application start. So, until this is not changed I'd keep throwing an error here.

Reply:
Ah okay, thanks for clarifying 👍